app/models/service_template_transformation_plan.rb

```diff
@@ -65,46 +66,20 @@ def self.create_catalog_item(options, _auth_user = nil)
     end
   end
 
-  private
-
-  def enforce_single_service_parent(_resource)
-  end
-
-  def self.validate_config_info(options)
-    config_info = options[:config_info]
-
-    mapping = if config_info[:transformation_mapping_id]
-                TransformationMapping.find(config_info[:transformation_mapping_id])
-              else
-                config_info[:transformation_mapping]
-              end
-
-    raise _('Must provide an existing transformation mapping') if mapping.blank?
-
-    pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
-    post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
+  def update_catalog_item(options, _auth_user = nil)
+    if miq_requests.any? || options[:config_info].nil?
+      options[:config_info] = self.options[:config_info]
+    end
 
-    vms = []
-    if config_info[:actions]
-      vm_objects = VmOrTemplate.where(:id => config_info[:actions].collect { |vm_hash| vm_hash[:vm_id] }.compact).index_by(&:id).stringify_keys
-      config_info[:actions].each do |vm_hash|
-        vm_obj = vm_objects[vm_hash[:vm_id]] || vm_hash[:vm]
-        next if vm_obj.nil?
+    added_vms_enhanced_config_info = validate_config_info(options)
 
-        vm_options = {}
-        vm_options[:pre_ansible_playbook_service_template_id] = pre_service_id if vm_hash[:pre_service]
-        vm_options[:post_ansible_playbook_service_template_id] = post_service_id if vm_hash[:post_service]
-        vms << {:vm => vm_obj, :options => vm_options}
+    transaction do
+      update_from_options(options)
+      vm_resources.destroy_all
+      reload.tap do |service_template|
+        added_vms_enhanced_config_info[:vms].each { |vm_hash| service_template.add_resource(vm_hash[:vm], :status => ServiceResource::STATUS_QUEUED, :options => vm_hash[:options]) }
+        service_template.create_resource_actions(added_vms_enhanced_config_info)
```

Comment:
do you need to `create_resource_actions` every time? If yes, shouldn't you delete them first?

Reply:
Do we have a method for deleting resource actions? If yes, then I think we should call it here.

Although note that we only want to delete the vm resources and rebuild them. 
As mentioned above https://github.com/ManageIQ/manageiq/pull/17989#discussion_r219525902, we don't want to update (and hence delete) the associated infrastructure mapping.